crates/shared/src/price_estimation.rs

```diff
@@ -179,6 +182,14 @@ pub struct Arguments {
     #[clap(long, env, default_value = "https://api.1inch.dev/")]
     pub one_inch_url: Url,
 
+    /// The API key for the CoinGecko API.
+    #[clap(long, env)]
+    pub coin_gecko_api_key: Option<String>,
```

Comment:
The autopilot configuration keeps growing with fields which are not really related to the autopilot. More reason to move this.